### PR TITLE
Add detekt task

### PR DESCRIPTION
**Pull request by benmcmahon100**
Add detekt for builds

### CODE REVIEW

**Review by yift-r3 — COMMENTED** [review]
Can we remove the TODO at line 8 in build.gradle.kts?

**Comment by benmcmahon100** [comment]
> Can we remove the TODO at line 8 in build.gradle.kts?

Done

**Comment by yift-r3** [comment]
In line 19, the detektPlugins("io.gitlab.arturbosch.detekt:detekt-formatting:1.12.0") should probably be 1.13.0

We have two detekt blocks (line 58 and 126)

**Comment by benmcmahon100** [comment]
> In line 19, the detektPlugins("io.gitlab.arturbosch.detekt:detekt-formatting:1.12.0") should probably be 1.13.0
> 
> We have two detekt blocks (line 58 and 126)

Good catch

**Review by yift-r3 — APPROVED** [review]

**Review by driessamyn — CHANGES_REQUESTED** [review]
This doesn't work.

File: build.gradle.kts [hunk 1]
```diff
@@ -1,13 +1,15 @@
...
+val rootProjectDir = rootDir
+
 
 plugins {
     kotlin("jvm") version "1.4.0"
 
-    // TODO: 1.13.0 was published today!. once jcentral get synch we should change the version
-    //       and set   warningsAsErrors: false in the config
-    id("io.gitlab.arturbosch.detekt") version "1.13.0"
+    id("io.gitlab.arturbosch.detekt") version "1.12.0"
```
Comment (yift-r3) [inline build.gradle.kts:12]: Why not 13?

File: build.gradle.kts [hunk 2]
```diff
@@ -87,26 +82,40 @@ subprojects {
...
+
+        }
+
+        detekt {
+            baseline = file("$rootProjectDir/detekt-baseline.xml")
+            config = files("$rootProjectDir/detekt-config.yml, " +
+                    "$rootProjectDir/detekt-baseline-config.yml")
+        }
 
+        // Added to support junit5 tests
```
Comment (yift-r3) [inline build.gradle.kts:113]: remove comment

File: build.gradle.kts (hunk 1 above)
Comment (benmcmahon100) [reply build.gradle.kts:12]: It can't resolve a load of dependencies if I bump it to 13

Comment (yift-r3) [reply build.gradle.kts:12]: :( Can you create a Jira ticket?

Comment (benmcmahon100) [reply build.gradle.kts:12]: Yeah sure thing

Comment (benmcmahon100) [reply build.gradle.kts:12]: https://r3-cev.atlassian.net/browse/INFRA-771

**Review by yift-r3 — APPROVED** [review]

**Review by yift-r3 — APPROVED** [review]

**Review by yift-r3 — APPROVED** [review]

File: build.gradle.kts [hunk 3]
```diff
@@ -81,30 +71,56 @@ subprojects {
...
+                    destination = file("$projectDir/build/detekt-report.xml")
+                }
+                html {
+                    enabled = true
+                    destination = file("$projectDir/build/detekt-report.html")
+                }
+                txt {
+                    enabled = true
+                    destination = file("$projectDir/build/detekt-report.txt")
+                }
```
Comment (driessamyn) [inline build.gradle.kts:121]: I don't think we need all 3 report types. We just need a machine readable and a human readable.
Currently, I don't think the human readable one is exposed in Jenkins.

**Pull request merged**